resource_migrations/run_all.py

```diff
@@ -0,0 +1,67 @@
+import argparse
+import os
+
+import migration_1
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-n",
+        "--namespace",
+        required=False,
+        default=os.environ.get("K8S_NAMESPACE", "default"),
+        type=str,
+        help="The k8s namespace where to run.",
+    )
+    parser.add_argument(
+        "-d",
+        "--dry-run",
+        action="store_true",
+        required=False,
+        help="If set, no changes are made at all.",
+    )
+    parser.add_argument(
+        "-g",
+        "--group",
+        type=str,
+        required=False,
+        default=os.environ.get("CRD_GROUP", "amalthea.dev"),
+        help="The group for the jupyterserver CRD.",
+    )
+    parser.add_argument(
+        "-a",
+        "--api-version",
+        type=str,
+        required=False,
+        default=os.environ.get("CRD_VERSION", "v1alpha1"),
+        help="The api version for the jupyterserver CRD.",
+    )
+    parser.add_argument(
+        "-p",
+        "--plural",
+        type=str,
+        required=False,
+        default=os.environ.get("CRD_PLURAL", "jupyterservers"),
+        help="The plural name for the jupyterserver CRD.",
+    )
+    parser.add_argument(
+        "--prefix",
+        type=str,
+        required=False,
+        default="renku.io/",
+        help="The renku k8s annotation prefix.",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def run_all(args):
+    # Run all migrations in order
+    print("Starting k8s resource migrations.")
+    migration_1.adjust_annotations(args)
```

Comment:
How about adding the "migration version" as a label to each notebook server? Then the `k8s_api.list_namespaced_custom_object` query could filter by this label and thus only select the objects which still need to apply the current migration.

Reply:
I added a migration version. This allows for a few things:
- migrations can run exactly on the right resources
- the migration version is part of the notebook helm chart - so now when new notebooks are created the migration version is already part of resource - so that the oldest migrations do not always run